deltacat/utils/placement.py

```diff
@@ -19,6 +20,15 @@
 # Issue: https://github.com/ray-project/ray/issues/29959
 
 
+def _get_available_resources_per_node():
+    # This API changed after this commit
```

Comment:
> Any way we can avoid using the private API? 

Fortunately for us, starting ray version 2.10.0, the API is no longer in private module and is annotated as DeveloperAPI. However, I understand that using private API must be discouraged and we should limit their usage.

Reply:
The description of DeveloperAPI is slightly unhelpful: 
"Developer APIs are lower-level methods explicitly exposed to advanced Ray users and library developers. **Their interfaces may change across minor Ray releases**."

But probably the fact that they decided to annotate it means they'd be at least somewhat more careful when changing it.
Anyways, the test should catch breakages -- good call to to add that!

Might be good to add a comment in addition to the link, so that the next Dmitri reading this doesn't complain.

Reply:
Sure, will add.